extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/indexing/error/MSQFilteredEmitterWarningPublisher.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.msq.indexing.error;
+
+import org.apache.druid.java.util.emitter.core.Emitter;
+import org.apache.druid.java.util.emitter.core.Event;
+import org.apache.druid.java.util.emitter.core.EventMap;
+
+import javax.annotation.Nullable;
+import java.io.IOException;
+import java.util.Set;
+
+public class MSQFilteredEmitterWarningPublisher implements MSQWarningPublisher
```

Comment:
I think it might be appropriate to have separate classes:
1. MSQFilteredPublisher which takes in a delegate and publishes the warning only if it passes through the filter
2. MSQEmitterPublisher that emits the warning if `publishException` is called

Reply:
That makes sense, but this is the only consumer of filtering and emitting right now so does that make sense to do that level of abstraction?

Reply:
If this is necessary i'll leave this change to later as it is pretty straightforward

Reply:
Works for me 👍